startblog/views.py

```diff
@@ -13,39 +13,65 @@
 from django.core.mail import send_mail, BadHeaderError
 from django.template.loader import render_to_string
 from django.http import HttpResponse
-from django.db.models import Exists,OuterRef,Count,Subquery
+from django.db.models import Exists, OuterRef, Count, Subquery
+from .tasks import *
+
 # Create your views here.
 
+
 def home(request):
     if not request.user.is_authenticated:
-        return redirect('blogapp:login')
-    blog = Blogs.objects.annotate ( is_liked = Exists(Likes.objects.filter(liked_user=request.user,post__id=OuterRef('pk'))),count =Count('blog_likes')).all()
-    context = {'blog':blog}
-    return render (request,'startblog/bloghome.html',context)
+        return redirect('blogapp:user_login')
+    '''annotate is used to add extra attribute data to the object'''
+    blog = Blogs.objects.annotate (is_liked = Exists(Likes.objects.filter(liked_user=request.user,post__id=OuterRef('pk'))),  
+    count =Count('blog_likes')).all()
+    context = {'blog': blog}
+    return render (request, 'startblog/bloghome.html', context)
+
 
 def all_blog(request):
     blog = Blogs.objects.all()
-    context = {'blog':blog}
-    return render(request,'startblog/all-blog.html',context)   
+    context = {'blog': blog}
+    return render(request,'startblog/all-blog.html', context)   
+
 
 def add_blog(request):
+    '''sub_list is subscribers for this particular user's posts '''
+    sub_list = Subscriber.objects.filter(user=request.user)
     if not request.user.is_authenticated:
-        return redirect('blogapp:login')
+        return redirect('blogapp:user_login')
     form = BlogsForm()   
-    initial_dict ={'bloger':request.user}
-    form = BlogsForm(request.POST or None ,initial=initial_dict)
-    if request.method=='POST':      
-        form = BlogsForm(request.POST,request.FILES,initial=initial_dict)       
+    initial_dict ={'bloger': request.user}
+    form = BlogsForm(request.POST or None, initial=initial_dict)
+    if request.method == 'POST':      
+        form = BlogsForm(request.POST,request.FILES, initial=initial_dict)       
         if form.is_valid():
             obj = form.save();
+            for subscriber in sub_list:
+                subject = 'New blog Update'
+                email_template_name = 'startblog/subscriber-mail.txt'
+                details = {
+                    'domain':'127.0.0.1:8000',
```

Comment:
add domain in settings.py and use that value

Reply:
Done